Framework/API/inc/MantidAPI/ExperimentInfo.h

```diff
@@ -163,12 +163,14 @@ class MANTID_API_DLL ExperimentInfo {
                                            const std::string &date = "");
 
   const Geometry::DetectorInfo &detectorInfo() const;
+  const Geometry::DetectorInfo &detectorInfoPhysical() const;
```

Comment:
Adding two more methods to the already huge `MatrixWorkspace`, is it necessary, given that this is used basically nowhere apart form once in `InstrumentView`?

Reply:
Good point.

Reply:
Though providing this addition does get us closer to the possibility that we don't need to store the Physical Instrument at all. Creating the `Instrument 2` bits in a lazy way at the point of use, means keeping the Physical Instrument.